esp32/modesp.c

```diff
@@ -37,6 +38,47 @@
 #include "drivers/dht/dht.h"
 #include "modesp.h"
 
+#define MODESP_INCLUDE_CONSTANTS (1)
+
+STATIC mp_obj_t esp_osdebug(size_t n_args, const mp_obj_t *args) {
+    switch (n_args) {
+    case 1:
+        if (args[0] == mp_const_none) {
+            esp_log_level_set("*", ESP_LOG_ERROR);
+        } else {
+            esp_log_level_set("*", LOG_LOCAL_LEVEL);
+        }
+        break;
+
+    case 2:
+        esp_log_level_set("*", mp_obj_get_int(args[1]));
+        break;
+    }
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_osdebug_obj, 1, 2, esp_osdebug);
+
+STATIC mp_obj_t esp_oslog(mp_obj_t level_t, mp_obj_t tag_o, mp_obj_t message_o) {
+    size_t tag_l, message_l;
+    const char *tag = mp_obj_str_get_data(tag_o, &tag_l);
+    const char *message = mp_obj_str_get_data(message_o, &message_l);
+    mp_uint_t level = mp_obj_get_int(level_t);
+
+    switch (level) {
+    case ESP_LOG_ERROR:     ; ESP_LOGE(tag, "%s\n", message); break;
+    case ESP_LOG_WARN:      ; ESP_LOGW(tag, "%s\n", message); break;
+    case ESP_LOG_INFO:      ; ESP_LOGI(tag, "%s\n", message); break;
+    case ESP_LOG_DEBUG:     ; ESP_LOGD(tag, "%s\n", message); break;
+    case ESP_LOG_VERBOSE:   ; ESP_LOGV(tag, "%s\n", message); break;
+    default:
+        mp_raise_msg(&mp_type_ValueError, "oslog: Invalid log level");
+    }
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(esp_oslog_obj, esp_oslog);
```

Comment:
You're absolutely right... there are other ways of delivering text. I think that use cases for print and logging are different though.  I like the way that (regular) Python at least allows for different types of endpoints. If the endpoint is the console, then it occurred to me (after I'd added the thing for testing) that it might be less jarring to present log output in a manner consistent with the OS (in terms of tags, colors, and so on). 

Again, though, I'll leave you